lightwood/mixers/sklearn_mixer.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+
+from sklearn.svm import SVC
+from sklearn.linear_model import SGDClassifier, SGDRegressor, LinearRegression
+from sklearn.naive_bayes import GaussianNB
+from sklearn.metrics import r2_score, balanced_accuracy_score
+
+from lightwood.logger import log
+from lightwood.constants.lightwood import COLUMN_DATA_TYPES
+from lightwood.mixers import BaseMixer
+
+CLASSIFICATION_MODELS = [
+    (SGDClassifier, {}),
+    (SVC, {}),
+    (GaussianNB, {}),
+]
+
+REGRESSION_MODELS = [
+    (LinearRegression, {}),
+    (SGDRegressor, {}),
+]
+
+class SklearnMixer(BaseMixer):
+    def __init__(self):
+        super().__init__()
+        self.targets = None
+        self.transformer = None
+        self.encoders = None
+
+    def fit(self, train_ds, test_ds=None):
+        """
+        :param train_ds: DataSource
+        :param test_ds: DataSource
+        """
+        self.fit_data_source(train_ds)
+
+        self.targets = {}
+
+        if test_ds is None:
+            test_ds = train_ds.subset(0.1)
+        else:
+            self.fit_data_source(test_ds)
+            assert test_ds.transformer is train_ds.transformer
+            assert test_ds.encoders is train_ds.encoders
+
+        self.transformer = train_ds.transformer
+        self.encoders = train_ds.encoders
+        
+        for output_feature in train_ds.output_features:
+            self.targets[output_feature['name']] = {
+                'type': output_feature['type']
+            }
+            if 'weights' in output_feature:
+                self.targets[output_feature['name']]['weights'] = output_feature['weights']
+            else:
+                self.targets[output_feature['name']]['weights'] = None
+
+        X_train = []
+        for row in train_ds:
+            X_train.append(np.array(row[0]))
+
+        X_test = []
+        for row in test_ds:
+            X_test.append(np.array(row[0]))
+
+        for target_col_name in self.targets:
+            Y_train = train_ds.get_column_original_data(target_col_name)
+            Y_test = test_ds.get_column_original_data(target_col_name)
```

Comment:
Is it ideal to do this regardless of what type our target is ? I.e.  numeric I'd assume it's easier to predict the min-maxed numbers.

For categorical... aren't the sklearn models able to predict OHE targets ?

Reply:
boost mixer uses get_column_original_data so i used it for sklearn mixer too
we can make an issue and change it for both mixers in a separate PR

Reply:
Agree